src/commands/importDocuments.ts

```diff
@@ -0,0 +1,127 @@
+    if (!uris) {
+        uris = await askForDocuments();
+    }
+    let ignoredUris: vscode.Uri[] = []; //account for https://github.com/Microsoft/vscode/issues/59782
+    uris = uris.filter((uri) => {
+        if (uri.fsPath.endsWith('.json')) {
+            return true;
+        } else {
+            ignoredUris.push(uri);
+            return false;
+        }
+    });
+    if (ignoredUris.length) {
+        ext.outputChannel.appendLine(`The following selected files are not json: ${ignoredUris.map(uri => uri.fsPath).join(',')}. \nWe are ignoring these.`);
+        ext.outputChannel.show();
+    }
+    const documents = await withProgress(parseDocumentsForErrors(uris), "Parsing documents...", vscode.ProgressLocation.Notification);
+
+    if (!collectionNode) {
+        collectionNode = <IAzureParentNode<MongoCollectionTreeItem | DocDBCollectionTreeItem>>await tree.showNodePicker([MongoCollectionTreeItem.contextValue, DocDBCollectionTreeItem.contextValue]);
+    }
+    let result: string;
+    if (collectionNode.treeItem instanceof MongoCollectionTreeItem) {
+        const collectionTreeItem = <MongoCollectionTreeItem>collectionNode.treeItem;
+        result = await collectionTreeItem.executeCommand('insertMany', [JSON.stringify(documents)]);
+    } else {
+        result = await insertDocumentsIntoDocdb(<IAzureParentNode<DocDBCollectionTreeItem>>collectionNode, documents, uris);
+    }
+    await collectionNode.refresh();
+    await vscode.window.showInformationMessage(result);
+}
+
+async function askForDocuments(): Promise<vscode.Uri[]> {
+    let files: vscode.Uri[] = await vscode.workspace.findFiles("*.json");
+    let jsonDocuments: (vscode.QuickPickItem & { uri: vscode.Uri })[] = [];
+    let items: (vscode.QuickPickItem & { uri: vscode.Uri })[] = files.map(file => {
+        return { uri: file, label: vscode.workspace.asRelativePath(file) };
+    });
+    let pickAgain: string = "Pick again";
+    let discontinue = "Discontiue import";
+    while (!jsonDocuments.length) {
+        jsonDocuments = await ext.ui.showQuickPick(items, { canPickMany: true, placeHolder: "Choose a document to upload. Hit Escape to Cancel" });
+        if (!jsonDocuments.length) {
+            let action: string = await vscode.window.showWarningMessage("No document picked. Want to pick again?", pickAgain, discontinue);
+            if (action === discontinue) {
+                throw new UserCancelledError();
+            }
+        }
+    }
+    return jsonDocuments.map(choice => choice.uri);
+}
+
+// tslint:disable-next-line:no-any
+async function parseDocumentsForErrors(nodes: vscode.Uri[]): Promise<any[]> {
+    const parseResult = await parseDocuments(nodes);
+    const documents = parseResult[0];
+    const errors: string[] = parseResult[1];
+    if (errors.length > 0) {
+        throw new Error(`Errors found in the following documents: ${errors.join(',')}.\nPlease fix these and try again.`);
```

Comment:
I wouldn't list out all errors here. It's probably way too much text. I would just direct them to the output channel if they need more details